packages/lib/service/attribute/server/getAttributes.ts

```diff
@@ -228,6 +144,188 @@ export async function getTeamMembersWithAttributeOptionValuePerAttribute({ teamI
+    orgId,
+    teamId,
+  });
+
+  const orgMembershipToUserIdForTeamMembers = new Map<OrgMembershipId, UserId>();
+
+  /**
+   * For an organization with 3000 users and 10 teams, with every team having around 300 members, the total memberships we query from DB are 3000+300 = 3300
+   * So, these are not a lot of records and we could afford to do in memory computations on them.
+   *
+   */
+  teamMemberships.forEach((teamMembership) => {
+    const orgMembership = orgMemberships.find(
+      (orgMembership) => orgMembership.userId === teamMembership.userId
+    );
+    if (!orgMembership) {
+      console.error(
+        `Org membership not found for userId ${teamMembership.userId} in the organization's memberships`
+      );
+      return;
+    }
+    orgMembershipToUserIdForTeamMembers.set(orgMembership.id, orgMembership.userId);
+  });
+
+  return orgMembershipToUserIdForTeamMembers;
+}
+
+async function _queryAllData({ orgId, teamId }: { orgId: number; teamId: number }) {
+  // Get all the attributes with their options first.
+  const [orgMembershipToUserIdForTeamMembers, attributesOfTheOrg] = await Promise.all([
+    _getOrgMembershipToUserIdForTeam({ orgId, teamId }),
+    AttributeRepository.findManyByOrgId({ orgId }),
+  ]);
+
+  const orgMembershipIds = Array.from(orgMembershipToUserIdForTeamMembers.keys());
+
+  // Get all the attributes assigned to the members of the team
+  const attributesToUsersForTeam = await AttributeToUserRepository.findManyByOrgMembershipIds({
+    orgMembershipIds,
+  });
+
+  return {
+    attributesOfTheOrg,
+    attributesToUsersForTeam,
+    orgMembershipToUserIdForTeamMembers,
+  };
+}
+
+async function getAttributesAssignedToMembersOfTeam({ teamId, userId }: { teamId: number; userId?: number }) {
+  const log = logger.getSubLogger({ prefix: ["getAttributeToUserWithMembershipAndAttributes"] });
+
+  const whereClauseForAttributesAssignedToMembersOfTeam = {
+    options: {
+      some: {
+        assignedUsers: {
+          some: {
+            member: {
+              userId,
+              user: {
+                teams: {
+                  some: {
+                    teamId,
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  log.debug(
+    safeStringify({
+      teamId,
+      whereClauseForAttributesAssignedToMembersOfTeam,
+    })
+  );
+
+  const assignedAttributeOptions = await prisma.attribute.findMany({
+    where: whereClauseForAttributesAssignedToMembersOfTeam,
+    select: {
+      id: true,
+      name: true,
+      type: true,
+      options: {
+        select: {
+          id: true,
+          value: true,
+          slug: true,
+          contains: true,
+          isGroup: true,
+        },
+      },
+      slug: true,
+    },
+  });
+
+  return assignedAttributeOptions;
+}
+
+export async function getAttributesAssignmentData({ orgId, teamId }: { orgId: number; teamId: number }) {
```

Comment:
This is the new export used in team members matching logic and thus has to be performant.